chromium_src/ui/views/controls/button/md_text_button.cc

```diff
@@ -39,11 +141,77 @@ class BraveTextButtonHighlightPathGenerator
 
 namespace views {
 
-// To be called from MdTextButtonBase::UpdateColors().
-void MdTextButtonBase::UpdateColorsForBrave() {
+MdTextButton::MdTextButton(PressedCallback callback,
+                           const std::u16string& text,
+                           int button_context)
+    : MdTextButtonBase(std::move(callback), text, button_context) {
+  SetCornerRadius(100);
+  views::HighlightPathGenerator::Install(
+      this, std::make_unique<BraveTextButtonHighlightPathGenerator>());
+  auto* ink_drop = views::InkDrop::Get(this);
+  views::InkDrop::UseInkDropForFloodFillRipple(ink_drop,
+                                               /*highlight_on_hover=*/false,
+                                               /*highlight_on_focus=*/true);
+  ink_drop->SetCreateHighlightCallback(base::BindRepeating(
+      [](Button* host) {
+        const SkColor fill_color = SK_ColorTRANSPARENT;
+        gfx::RectF boundsF(host->GetLocalBounds());
+        return std::make_unique<InkDropHighlight>(
+            boundsF.size(), static_cast<MdTextButton*>(host)->GetCornerRadius(),
+            boundsF.CenterPoint(), fill_color);
+      },
+      this));
+}
+
+MdTextButton::~MdTextButton() = default;
+
+SkPath MdTextButton::GetHighlightPath() const {
+  SkPath path;
+  int radius = GetCornerRadius();
+  path.addRRect(
+      SkRRect::MakeRectXY(RectToSkRect(GetLocalBounds()), radius, radius));
+  return path;
+}
+
+MdTextButton::Kind MdTextButton::GetKind() const {
```

Comment:
I don't understand why we're adding `Kind` here or how it's different from `Style`. Why do we have something completely different here instead of just mapping the upstream styles to whatever we want?

Reply:
At that time, we didn't change upstream's button color but we wanted to use different colors with `MdTextButton` so another style (`Kind`) was needed.